### PR TITLE
Cart screen with functionality

### PR DESCRIPTION
**Pull request by dev-enforced**
In this PR one can find the cart feature of the Eleganz Store where one can add the products into the cart after logging into the application. 
So you can go to the products page and if you click on add to cart on any of the products then it will ask you to login if not logged in and if you are logged in then the product can be added to cart after which you will be given an option to move to the cart where you can check the status of the products in cart. If the cart gets empty then options can be shown to add items into the cart.
Was unable to break this into another cart context file so tried to do this using the existing authentication-context. so if anyone can help out in splitting into multiple contexts do connect!!
Live preview Link for trying out this feature: https://deploy-preview-9--eleganz-store.netlify.app/

### CODE REVIEW

File: src/pages/Cart/Cart.css [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+/* FILLED CART CONTAINER */
+
+.filled-cart-content {
+    flex-grow: 1;
+    background-color: var(--bg-color);
+}
+
+.cart-summary-container{
+    display: grid;
+    grid-template-columns: 1fr 1fr;
```
Comment (SaurabhChawda) [inline src/pages/Cart/Cart.css:24]: Liked that you are using Grid.

**Review by SaurabhChawda — APPROVED** [review]

**Comment by SaurabhChawda** [comment]
Design looks good to me, You have used well-maintained color but In the login and signup form design looks a little big you can reduce the size of the form. (personal opinion). Apart from that everything looks good to me.You can merge it.

File: src/components/CartPriceDetails/CartPriceDetails.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { useAuthentication } from 'context';
+import React from 'react';
+import "./CartPriceDetails.css";
+
+const CartPriceDetails = () => {
+    const { authState: { cart: cartProvided } } = useAuthentication();
+    const originalPriceSum = cartProvided.reduce((total, everyCartItem) => total += Number(everyCartItem.originalPrice), 0)
+    const discountedPriceSum = cartProvided.reduce((total, everyCartItem) => total += Number(everyCartItem.discountedPrice), 0)
```
Comment (JantuDeb) [inline src/components/CartPriceDetails/CartPriceDetails.jsx]: You can do this in one `reduce` call, return an object which will contain original and discount
```
const  priceSum = cartProvided.reduce((acc, everyCartItem) =>{
return ({...acc, original: Number(everyCartItem.originalPrice) +acc.original, discount:Number(everyCartItem.discountedPrice)+acc.discount})
} , {original:0, discount:0})
```

**Review by JantuDeb — APPROVED** [review]
Overall it is nicely coded. I have added a comment do check it out

File: src/components/CartPriceDetails/CartPriceDetails.jsx (hunk 2 above)
Comment (dev-enforced) [reply src/components/CartPriceDetails/CartPriceDetails.jsx]: Thanks Jantu for pointing this out !! Will implement it right away . Got a mini issue to be addressed that got solved while using the quantity attribute of every item in the cart

**Pull request merged**